plugins/inputs/gtfs/gtfs.go

```diff
@@ -0,0 +1,305 @@
+			"json":          b,
+		}
+
+		tags := map[string]string{
+			"route_id":     *v.Trip.RouteId,
+			"direction_id": fmt.Sprintf("%d", v.Trip.DirectionId),
+		}
+
+		acc.AddFields("position", fields, tags, time.Unix(int64(*v.Timestamp), 0))
+	}
+
+	return nil
+}
+
+func (g *GTFS) gatherTripUpdates(acc telegraf.Accumulator, t time.Time) error {
+	entities, err := g.gather(g.tripUpdatesReq)
+	if err != nil {
+		return err
+	}
+
+	for _, entity := range entities {
+		if entity.TripUpdate == nil {
+			continue
+		}
+		u := entity.TripUpdate
+
+		b, err := json.Marshal(u)
+		if err != nil {
+			return err
+		}
+
+		for _, update := range u.StopTimeUpdate {
+			fields := map[string]interface{}{
+				"vehicle_id":    u.Vehicle.Id,
+				"vehicle_label": u.Vehicle.Label,
+				"trip_id":       u.Trip.TripId,
+				"stop_id":       update.StopId,
+				"stop_sequence": update.StopSequence,
+				"arrival":       update.Arrival.Time,
+				"departure":     update.Departure.Time,
+				"delay":         u.Delay,
+				"json":          b,
+			}
+
+			tags := map[string]string{
+				"route_id":     *u.Trip.RouteId,
+				"direction_id": fmt.Sprintf("%d", u.Trip.DirectionId),
+			}
+
+			acc.AddFields("update", fields, tags, time.Unix(int64(*u.Timestamp), 0))
+		}
+	}
+
+	return nil
+}
+
+func (g *GTFS) gatherServiceAlerts(acc telegraf.Accumulator, t time.Time) error {
+	entities, err := g.gather(g.serviceAlertsReq)
+	if err != nil {
+		return err
+	}
+
+	for _, entity := range entities {
+		if entity.Alert == nil {
+			continue
+		}
+		a := entity.Alert
+
+		b, err := json.Marshal(a)
+		if err != nil {
+			return err
+		}
+
+		fields := map[string]interface{}{
+			"cause": a.Cause.String(),
+			"json":  b,
+		}
+
+		tags := map[string]string{
+			"severity": a.SeverityLevel.String(),
+		}
+
+		acc.AddFields("alert", fields, tags, time.Now())
+	}
+
+	return nil
+}
+
+func (g *GTFS) gather(req *http.Request) ([]*gtfs.FeedEntity, error) {
+	resp, err := g.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var feed gtfs.FeedMessage
+	err = proto.Unmarshal(body, &feed)
+	if err != nil {
+		return nil, err
+	}
+
+	return feed.Entity, nil
+}
+
+func (g *GTFS) newRequest(s string) (*http.Request, error) {
+	u, err := url.Parse(s)
+	if err != nil {
+		return nil, err
+	}
+
+	if g.Key != "" {
+		q := u.Query()
+		q.Set("key", g.Key)
+		u.RawQuery = q.Encode()
+	}
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if g.Username != "" && g.Password != "" {
+		req.SetBasicAuth(g.Username, g.Password)
+	}
+
+	return req, nil
+}
+
+func init() {
+	inputs.Add("gtfs", func() telegraf.Input { return &GTFS{} })
+}
+
+var errNoSourceConfigured = errors.New("No source url configured; at least one of vehicle_positions_url, trip_updates_url or service_alerts_url must be populated")
```

Comment:
Please remove this global variable and move the error to the only place where you use it.